src/instructlab/sdg/pipeline.py

```diff
@@ -69,3 +81,60 @@ def generate(self, dataset) -> Dataset:
                 dataset = self._drop_duplicates(dataset, cols=drop_duplicates_cols)
 
         return dataset
+
+
+_block_types = {
+    "CombineColumnsBlock": utilblocks.CombineColumnsBlock,
+    "ConditionalLLMBlock": llmblock.ConditionalLLMBlock,
+    "FilterByValueBlock": filterblock.FilterByValueBlock,
+    "LLMBlock": llmblock.LLMBlock,
+    "SamplePopulatorBlock": utilblocks.SamplePopulatorBlock,
+    "SelectorBlock": utilblocks.SelectorBlock,
+}
+
+
+def _lookup_block_type(block_type):
+    if not block_type in _block_types:
+        raise FlowParserError("Unknown block type {block_type}")
+    return _block_types[block_type]
+
+
+_FLOW_PARSER_MAJOR = 1
+_FLOW_PARSER_MINOR = 0
+
+
+class FlowParserError(Exception):
+    """An exception raised while parsing a flow config file."""
+
+
+def parse_flow_config_file(flow_path):
+    with open(flow_path, "r", encoding="utf-8") as flow_file:
+        content = yaml.safe_load(flow_file)
+
+    version = content["version"]
+    major, minor = map(int, version.split("."))
+
+    if major > _FLOW_PARSER_MAJOR:
+        raise FlowParserError(
+            "The custom flow file format is from a future major version."
+        )
+    if major <= _FLOW_PARSER_MAJOR and minor > _FLOW_PARSER_MINOR:
+        logger.warning(
+            "The custom flow file may have new features that will be ignored."
+        )
+
+    if not "block_configs" in content:
+        raise FlowParserError(
+            "The custom flow file contains no 'block_configs' section"
+        )
```

Comment:
This is not a blocking thing, but a nice minor improvement to these Execption messages would be to include the specified version from the file we loaded as well as the latest version the code understands